test/distributed/org/apache/cassandra/distributed/test/log/SystemKeyspaceStorageTest.java

```diff
@@ -119,12 +118,16 @@ public void testLogStateQuery() throws Throwable
                         }
                         else
                         {
-                            assertEquals(latestSnapshot, logState.baseState.epoch);
+                            assertEquals(since, logState.baseState.epoch);
                             start = logState.baseState.epoch;
                             if (logState.entries.isEmpty()) // no entries, snapshot should have the same epoch as since
                                 assertEquals(since, start);
                             else // first epoch in entries should be snapshot epoch + 1
+                            {
+                                if (!start.nextEpoch().equals(logState.entries.get(0).epoch))
```

Comment:
Is this supposed to stay in?

Reply:
No, this was my oversight, thanks for catching!